### PR TITLE
Lobby UI rework

### PR DESCRIPTION
**Pull request by NoooneyDude**
- Moves some code from django branch.
    - Improves startup -> login flow spaghetti, including less `successString` / `errorString` callbacks.
- Creates reusable loading and info panels for the lobby UI.
- Moves business logic out of the UI code.
- Small UI/UX improvements.
- Simple client form validations.
- More logging for player spawn.
- Moves more code to relevant scripts: the main menu, account login, account creation server join and server history.
- Splits the `LoggedOut` event into one for logging out of the server and and one for logging out of your account (they were used interchangeably :eyes: (though never invoked))

### CODE REVIEW

**Comment by NoooneyDude** [comment]
Alright that'll do for now, ready for review

File: UnityProject/Assets/Scripts/Core/Chat/DiscordWebhookMessage.cs [hunk 1]
```diff
@@ -251,10 +251,18 @@ void HandleLog(string logString, string stackTrace, LogType type)
 		{
 			if (type == LogType.Exception || type == LogType.Error)
 			{
-				GameManager.Instance.errorCounter++;
-				if (ErrorMessageHashSet.Contains(stackTrace))
-					return;
-				GameManager.Instance.uniqueErrorCounter++;
+				bool isUnique = ErrorMessageHashSet.Contains(stackTrace);
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Core/Chat/DiscordWebhookMessage.cs]:
```suggestion
				bool isUnique = ErrorMessageHashSet.Contains(stackTrace) == false;
```

Comment (corp-0) [reply UnityProject/Assets/Scripts/Core/Chat/DiscordWebhookMessage.cs]: github suggestion isn't an IDE!

**Comment by Fr-Dae** [comment]
Job2
**like that ?**

**Comment by corp-0** [comment]
no, this reworks the way the lobby scene works in the backend.The one you showed is the profession preferences screen.

**Comment by ThatDan123** [comment]
Do you want to ignore that codacy suggestion?

**Comment by NoooneyDude** [comment]
> Do you want to ignore that codacy suggestion?

I do, as the value could change again.

**Comment by NoooneyDude** [comment]
Eh, guess I should rearrange it 👍

**Review by Bod9001 — APPROVED** [review]
from earlier

**Pull request merged**